vars/buildPlugin.groovy

```diff
@@ -219,3 +219,21 @@ static List<Map<String, String>> getConfigurations(Map params) {
     }
     return ret
 }
+
+/**
+ * Get recommended configurations for testing.
+ * Includes testing Java 8 and 11 on the newest LTS.
+ */
+static List<Map<String, String>> recommendedConfigurations() {
```

Comment:
I'm sorry, I should have told you before, but I've been thinking about this, and I really think this function must take an argument so we can evolve it over time without breaking existing usages.

IOW, would become something like:

```groovy
static List<Map<String, String>> recommendedConfigurations(configDate) {
  if(version="2019-03-08") {
    //TODO: replace by 2.164.1 once it is released
    def recentLTS = "2.164"
    def configurations = [
        [ platform: "linux", jdk: "8", jenkins: null ],
        [ platform: "windows", jdk: "8", jenkins: null ],
        [ platform: "linux", jdk: "8", jenkins: recentLTS, javaLevel: "8" ],
        [ platform: "windows", jdk: "8", jenkins: recentLTS, javaLevel: "8" ],
        [ platform: "linux", jdk: "11", jenkins: recentLTS, javaLevel: "8" ],
        [ platform: "windows", jdk: "11", jenkins: recentLTS, javaLevel: "8" ]
    ]
    return configurations
  } else if ( version=...) {
    ...
  } else {
     throw new UnknownConfigException("blah");
  }
}
```

This way, we will be able to evolve the recommended config without suddenly breaking plugins builds out there.

WDYT?

Reply:
> evolve the recommended config without suddenly breaking plugins builds out there

I have long advocated that plugins not simply import the `master` version of this library, but only a specific version, so that updates to the library can be rolled out in a controlled fashion in tested PRs. (With @dependabot, even!) @rtyler never did so; not sure if he specifically argued against it, or just did not want to spend the time.